src/treemodel/delta/delta.js

```diff
@@ -128,4 +167,64 @@ export default class Delta {
 	static get _priority() {
 		return 0;
 	}
+
+	/**
+	 * Creates InsertDelta from deserialized object, ie. from parsed JSON string
+	 *
+	 * @param {Object} json
+	 * @param {engine.treeModel.Document} doc Document on which this delta will be applied.
+	 * @returns {engine.treeModel.delta.InsertDelta}
+	 */
+	static fromJSON( json, doc ) {
+		if ( !deserializers.has( json.__className ) ) {
+			/**
+			 * This delta has no defined deserializer.
+			 *
+			 * @error delta-fromjson-no-deserializer
+			 * @param {String} name
+			 */
+			throw new CKEditorError(
+				'delta-fromjson-no-deserializer: This delta has no defined deserializer',
+				{ name: json.__className }
+			);
+		}
+
+		let Constructor = deserializers.get( json.__className );
+
+		let delta = new Constructor();
+
+		if ( json.operations.length ) {
+			json.operations.forEach( ( operation ) => delta.addOperation( operations[ operation.__className ].fromJSON( operation, doc ) ) );
+		}
+
+		return delta;
+	}
+}
+
+const operations = {};
+operations[ AttributeOperation.className ] = AttributeOperation;
+operations[ InsertOperation.className ] = InsertOperation;
+operations[ MoveOperation.className ] = MoveOperation;
+operations[ NoOperation.className ] = NoOperation;
+operations[ Operation.className ] = Operation;
+operations[ ReinsertOperation.className ] = ReinsertOperation;
+operations[ RemoveOperation.className ] = RemoveOperation;
```

Comment:
But we where thinking a lot about `OperationFactory`, which might be useful also to resolve circular dependencies between operations, so I think it might be useful for more then deserialisation`.